runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -6546,14 +6547,37 @@ done:;
 				restoreGenericSpecialStackFrame(REGISTER_ARGS);
 				if (immediateAsyncPending()) {
 					rc = GOTO_ASYNC_CHECK;
+					goto done;
 				} else if (VM_VMHelpers::exceptionPending(_currentThread)) {
 					rc = GOTO_THROW_CURRENT_EXCEPTION;
+					goto done;
 				} else {
 					profileCallingMethod(REGISTER_ARGS);
 					_sendMethod = method;
 				}
 			}
+
```

Comment:
It seems you're missing a bunch of the code from invokespecial here - fetching host class and seeing if it's an interface.